File: payroll-api/src/main/java/com/company/payroll/mapper/StaffPayrollMapper.java
package com.company.payroll.mapper;

import java.util.List;

import com.company.payroll.model.StaffPayroll;
import org.springframework.stereotype.Repository;

@Repository
public interface StaffPayrollMapper {
    int deleteByPrimaryKey(Integer prId);
    int insert(StaffPayroll row);
    int insertSelective(StaffPayroll row);
    List<StaffPayroll> selectList();
    List<StaffPayroll> selectListByStaffId(Integer staffId);
    StaffPayroll selectByPrimaryKey(Integer prId);
    int updateByPrimaryKeySelective(StaffPayroll row);
    int updateByPrimaryKey(StaffPayroll row);
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/mapper/StaffPayrollMapper.kt
package com.company.payroll.mapper

import com.company.payroll.model.StaffPayroll
import org.springframework.stereotype.Repository

@Repository
interface StaffPayrollMapper {
  fun deleteByPrimaryKey(prId: Int): Int

  fun insert(row: StaffPayroll): Int

  fun insertSelective(row: StaffPayroll): Int

  fun selectByPrimaryKey(prId: Int): StaffPayroll

  fun selectList(): List<StaffPayroll>

  fun selectListByStaffId(staffId: Int): List<StaffPayroll>

  fun updateByPrimaryKeySelective(row: StaffPayroll): Int

  fun updateByPrimaryKey(row: StaffPayroll): Int
}